File: cadastro-moradores-api/src/test/java/com/elbertribeiro/cadastromoradores/usuario/UsuarioServicoTest.java
package com.elbertribeiro.cadastromoradores.usuario;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;
import org.mockito.Mock;

import java.util.List;

import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;

class UsuarioServicoTest {

    @Mock
    List<Usuario> usuarios;
    Usuario usuario = mock(Usuario.class);
    UsuarioRepositorio repositorio = mock(UsuarioRepositorio.class);

    @Test
    void buscarListaUsuariosTest() {
        when(repositorio.findAll()).thenReturn(usuarios);
        UsuarioServico usuarioServico = new UsuarioServico(repositorio);
        Assertions.assertEquals(usuarios, usuarioServico.buscarListaUsuarios());
    }

    @Test
    void salvarUsuarioTest() {
        when(repositorio.save(usuario)).thenReturn(usuario);
        UsuarioServico usuarioServico = new UsuarioServico(repositorio);
        Assertions.assertNotNull(usuarioServico.salvarUsuario(usuario));
    }
}

File: cadastro-moradores-api/src/test/kotlin/com/elbertribeiro/cadastromoradores/usuario/UsuarioServicoTest.kt
package com.elbertribeiro.cadastromoradores.usuario

import io.mockk.every
import io.mockk.mockk
import io.mockk.verify
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

class UsuarioServicoTest {

    @Test
    fun buscarListaUsuarios() {
        val mockedList: MutableList<Usuario> = mutableListOf()

        val repositorio = mockk<UsuarioRepositorio>()
        every { repositorio.findAll() } returns mockedList

        val servico = UsuarioServico(repositorio)

        val result = servico.buscarListaUsuarios()
        verify { servico.buscarListaUsuarios() }

        assertEquals(mockedList, result)
    }
}